lib/rules/selector-max-id/index.mjs

```diff
@@ -66,15 +66,17 @@ const rule = (primary, secondaryOptions) => {
 			}, 0);
 
 			if (selectorNode.type !== 'root' && selectorNode.type !== 'pseudo' && count > primary) {
```

Comment:
minor
perf

You can put `if (selectorNode.type === 'root' || selectorNode.type === 'pseudo') return;` before l53 so that you don't call the `reduce` in those cases, ending up with

```js
if (count > primary)
```

Reply:
Nice find!

Would be nice to apply this everywhere the same pattern exists.

Reply:
I tried this and we can't make this change because `checkSelector` is recursive.